### DISCUSSION TITLE
How to install/enable tidy extension?

### DISCUSSION THREAD

**Discussion by arnor01** [beyondcode/herd-community#682] [Q&A] [answered]
I have a php7.4 application which uses Tidy.
I tried to install it via pecl as stated in the [docs](https://herd.laravel.com/docs/1/advanced-usage/additional-extensions) but that returns: `pear/tidy requires PHP (version >= 4.3.0, version <= 6.0.0), installed version is 7.4.33`.
Is there anyone who knows how to setup the tidy extension for herd on MacOS? Any help would be appreciated!

**Comment by finn-matti** [comment]
I think you can't. The php provided would've needed to be compiled with the `--with-tidy` option. As per [the PHP docs](https://www.php.net/manual/en/tidy.installation.php).

I also tried to follow this guide: https://atlas.dev/blog/add-extensions-to-laravel-herd-without-homebrew

And found (seemingly) the so file: https://packages.macports.org/php84-tidy/

But setting it in the php.ini like so:

```
extension=/path/to/tidy.so
```

didn't make a difference.

That said: Would be nice if it would just be included.

**Comment by finn-matti** [comment, marked as answer]
Actually, it works! I just put the so into the wrong ini file (I thought I was using php 8.4, but I was using php 8.3). Caveat: You will have to install macports and tidy through macports. So here's how it works:

1. Download and extract https://packages.macports.org/php84-tidy/ (php83-tidy or whatever)
2. Find the `tidy.so` in the folder `php83-tidy-8.3.12_0.darwin_25.arm64/opt/local/lib/php83/extensions/no-debug-non-zts-20230831/`
3. Copy that to a path, [the guide](https://atlas.dev/blog/add-extensions-to-laravel-herd-without-homebrew) I was using for this uses `/Users/<User>/Library/Application Support/Herd/config/php/83`
4. [Install macports](https://www.macports.org/install.php)
5. Install [Tidy](https://ports.macports.org/port/tidy/)
6. Put `extension=/path/to/tidy.so` into your php.ini

And then it works!

```
$ which php
/Users/<User>/Library/Application Support/Herd/bin//php
$ php -v
PHP 8.3.27 (cli) (built: Oct 24 2025 10:54:28) (NTS clang 15.0.0)
Copyright (c) The PHP Group
Zend Engine v4.3.27, Copyright (c) Zend Technologies
    with Xdebug v3.3.1, Copyright (c) 2002-2023, by Derick Rethans
    with Zend OPcache v8.3.27, Copyright (c), by Zend Technologies
$ php -i | grep Tidy
Tidy support => enabled
libTidy Version => 5.8.0
libTidy Release => 2021/07/10
```
